src/Core/Domain/Product/Exception/CannotUpdateProductException.php

```diff
@@ -31,6 +31,11 @@
  */
 class CannotUpdateProductException extends ProductException
 {
+    /**
+     * When generic product update fails
+     */
+    public const FAILED_UPDATE_PRODUCT = 0;
```

Comment:
Oh, I didn't think that we would replace all the other error codes by a common one, nice catch! Although it's probably better not to use the 0 value which is the default error code when nothing is specified.

```suggestion
    public const FAILED_UPDATE_PRODUCT = 1;
```